src/services/services.ts

```diff
@@ -2072,7 +2079,8 @@ namespace ts {
             compilationSettings: CompilerOptions,
             scriptSnapshot: IScriptSnapshot,
             version: string,
-            acquiring: boolean): SourceFile {
+            acquiring: boolean,
+            scriptKind?: ScriptKind): SourceFile {
 
             const bucket = getBucketForCompilationSettings(compilationSettings, /*createIfMissing*/ true);

```

Comment:
Question is: do we think if this scenario is valid? if no then we can just an extra check that file with some name  should be queried from the register using the same script kind all the time and move on. If yes then we need a way to distinguish such files.